plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidViewHolderRefactoring.java

```diff
@@ -0,0 +1,312 @@
+                    }
+                } else if (ancestor instanceof Assignment) {
+                    viewVariableAssignment = (Assignment) ancestor;
+                    ASTNode leftHandSideNode = viewVariableAssignment.getLeftHandSide();
+                    if (leftHandSideNode.getNodeType() == SIMPLE_NAME) {
+                        viewVariable = (SimpleName) leftHandSideNode;
+                    } else {
+                        resetData();
+                        return VISIT_SUBTREE;
+                    }
+                    viewAssignmentStmt = getAncestorOrNull(viewVariableAssignment, ExpressionStatement.class);
+                }
+                return DO_NOT_VISIT_SUBTREE;
+            }
+            return VISIT_SUBTREE;
+        }
+
+        public boolean visit(ReturnStatement node) {
+            this.returnStmt = node;
+            return VISIT_SUBTREE;
+        }
+
+        public boolean isInflateInsideIf() {
+            if (this.viewAssignmentStmt != null) {
+                Expression inflateExpr = getInflateExpr();
+                return getFirstAncestorOrNull(this.viewAssignmentStmt, IfStatement.class, SwitchStatement.class) != null
+                        // check whether inflate is inside a conditional
+                        // assignment
+                        || (inflateExpr != null && inflateExpr.getNodeType() == ASTNode.CONDITIONAL_EXPRESSION);
+            }
+            return false;
+        }
+
+        public Expression getInflateExpr() {
+            if (this.viewVariableDeclarationFragment != null) {
+                return this.viewVariableDeclarationFragment.getInitializer();
+            } else if (this.viewVariableAssignment != null) {
+                return this.viewVariableAssignment.getRightHandSide();
+            }
+            return null;
+        }
+
+        private boolean isInflateMethod(MethodInvocation node) {
+            return isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup")
+                    || isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup",
+                            "boolean")
+                    || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                            "android.view.ViewGroup")
+                    || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                            "android.view.ViewGroup", "boolean");
+        }
+    }
+
+    static class FindViewByIdVisitor extends ASTVisitor {
+        private List<FindViewByIdItem> items = new ArrayList<FindViewByIdItem>();
+
+        static class FindViewByIdItem {
+            private SimpleName variable;
+            private Expression findViewByIdExpr;
+            private VariableDeclarationFragment findViewByIdDeclarationFragment;
+            private Assignment findViewByIdVariableAssignment;
```

Comment:
(No need to act on this I fixed it locally)

`findViewByIdDeclarationFragment` and `findViewByIdVariableAssignment` are [singular fields](http://www.ing.iac.es/~docs/external/java/pmd/rules/controversial.html#SingularField).
They should be transformed into local variables.